appcues/src/main/java/com/appcues/action/ActionProcessor.kt

```diff
@@ -68,25 +75,21 @@ internal class ActionProcessor(override val scope: Scope) : KoinScopeComponent {
         actions: List<ExperienceAction>,
         interactionType: InteractionType,
         viewDescription: String?
-    ): StepInteractionAction? {
+    ): StepInteractionAction {
         // reverse the action list to get the first from back to front
-        return actions.reversed()
-            .firstOrNull { it is MetadataSettingsAction }
-            ?.let { action ->
-                with(action as MetadataSettingsAction) {
-                    // get an instance of StepInteractionData so it can access
-                    // StateMachine and AnalyticsTracker
-                    scope.get {
-                        parametersOf(
-                            StepInteractionData(
-                                interactionType = interactionType,
-                                viewDescription = viewDescription,
-                                category = category,
-                                destination = destination,
-                            )
-                        )
-                    }
-                }
-            }
+        val primaryAction = actions.reversed().firstOrNull { it is MetadataSettingsAction } as? MetadataSettingsAction
+
+        // get an instance of StepInteractionData so it can access
+        // StateMachine and AnalyticsTracker
+        return scope.get {
+            parametersOf(
+                StepInteractionData(
+                    interactionType = interactionType,
+                    viewDescription = viewDescription,
+                    category = primaryAction?.category ?: "",
+                    destination = primaryAction?.destination ?: "",
+                )
+            )
+        }
```

Comment:
simplified to always return the `StepInteractionAction`, even if no `MetadataSettingsAction` is found (it should always be...) - with default values in category and destination. Done for [consistency with how iOS is reporting](https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Presentation/Actions/ActionRegistry.swift#L79).